src/components/List.js

```diff
@@ -1,39 +1,42 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { FirestoreDocument } from 'react-firestore';
-import ListItem from './ListItem';
-import '../css/components/ItemsList.css';
+import SortedList from './SortedList';
 import { getToken } from '../lib/TokenService';
 import AddButton from './AddButton';
+import { filter as removePunctuation } from '../lib/helper';
+import '../css/components/ItemsList.css';
 
 export default function List() {
+  const [searchTerm, setSearchTerm] = useState('');
+
   return (
     <div>
-      <h1>LIST OF ITEMS</h1>
+      <h3>LIST OF ITEMS</h3>
       <FirestoreDocument
         path={`shoppingLists/${getToken()}`}
         render={({ isLoading, data }) => {
-          const itemsKeys = data
-            ? Object.keys(data).filter((key) => key !== 'id')
-            : '';
+          let itemsKeys = data ? getItemsKeys(data, searchTerm) : [];
           return isLoading ? (
             <div className="m-auto">Loading</div>
           ) : (
             <div>
-              {itemsKeys.length >= 1 ? (
-                <ul>
-                  {itemsKeys
-                    .sort((a, b) => (a > b ? 1 : -1))
-                    .map((key) => {
-                      return (
-                        <ListItem
-                          listItem={data[key]}
-                          key={key}
-                          listId={data.id}
-                          itemId={key}
-                        />
-                      );
-                    })}
-                </ul>
+              {itemsKeys.length || searchTerm ? (
+                <>
+                  <div className="search-term-wrapper">
+                    <input
+                      value={searchTerm}
+                      onChange={(e) => setSearchTerm(e.target.value.trim())}
+                      placeholder="Search item"
+                    />
+                    <button
+                      onClick={() => setSearchTerm('')}
+                      disabled={searchTerm ? false : true}
```

Comment:
I would prefer to use `hidden` here instead of `disabled` just a personal preference :D

Reply:
We were actually using it with something like `{searchTerm && theBtn}` but it felt a bit jumpy, like in the gif (yes, we were too lazy to make a new gif after changing the button 😅). Anyway, we have plenty of time for styling in later issues. We can change EVERYTHING then. I've already started to dislike how the app looks haha. 😸